packages/devtools_app/lib/src/framework/observer/memory_observer.dart

```diff
@@ -72,14 +82,76 @@ class MemoryObserver extends DisposableController {
       );
     }
   }
+
+  static Future<bool> _memoryExceedsThreshold({
+    @visibleForTesting Future<int?> Function()? debugMeasureUsageInBytes,
+  }) async {
+    final memoryUsageInBytes =
+        debugMeasureUsageInBytes != null
+            ? await debugMeasureUsageInBytes()
+            : await measureMemoryUsageInBytes();
+    _lastMemoryUsageInBytes = memoryUsageInBytes;
```

Comment:
I would just return this as a part of the response (could be a record), as opposed to using a long lived field to keep track of it. And then pass the starting memory usage into `reduceMemory` as a parameter.

Reply:
To pass it into `reduceMemory` as a parameter, we'd have to make an additional request to `measureMemoryUsageInBytes` to get the latest value. It is possible that the warning has been showing for a while before the user presses "Reduce memory", so we may not have the most up to date information at the time that we show the prompt. Updating a long lived field here ensures that we always have the latest value available. Calling `measureMemoryUsageInBytes` is not fast, so calling it again in an onPressed handler wouldn't be a great experience for performance.